packages/web/src/pages/chat-page/components/ChatListBlastItem.tsx

```diff
@@ -40,72 +19,12 @@ type ChatListBlastItemProps = {
 
 export const ChatListBlastItem = (props: ChatListBlastItemProps) => {
   const { chat, onChatClicked, currentChatId } = props
-  const {
-    chat_id: chatId,
-    audience,
-    audience_content_id: audienceContentId,
-    audience_content_type: audienceContentType
-  } = chat
+  const { chat_id: chatId } = chat
   const isCurrentChat = currentChatId && currentChatId === chatId
-  const decodedContentId = audienceContentId
-    ? decodeHashId(audienceContentId) ?? undefined
-    : undefined
-
-  const { data: currentUserId } = useGetCurrentUserId({})
-  const { data: user } = useGetCurrentUser()
-  const { data: track } = useGetTrackById(
-    {
-      id: decodedContentId!
-    },
-    { disabled: !audienceContentId || audienceContentType !== 'track' }
-  )
-  const { data: album } = useGetPlaylistById(
-    {
-      playlistId: decodedContentId!
-    },
-    { disabled: !audienceContentId || audienceContentType !== 'album' }
-  )
-
-  const { data: purchasersCount } = useGetPurchasersCount(
-    {
-      userId: currentUserId!,
-      contentId: decodedContentId,
-      contentType: audienceContentType
-    },
-    {
-      disabled: audience !== ChatBlastAudience.CUSTOMERS || !currentUserId
-    }
-  )
-  const { data: remixersCount } = useGetRemixersCount(
-    {
-      userId: currentUserId!,
-      trackId: decodedContentId
-    },
-    {
-      disabled: audience !== ChatBlastAudience.REMIXERS || !currentUserId
-    }
-  )
-
-  const audienceCount = useMemo(() => {
-    switch (audience) {
-      case ChatBlastAudience.FOLLOWERS:
-        return user?.follower_count
-      case ChatBlastAudience.TIPPERS:
-        return user?.supporter_count
-      case ChatBlastAudience.CUSTOMERS:
-        return purchasersCount
-      case ChatBlastAudience.REMIXERS:
-        return remixersCount
-      default:
-        return 0
-    }
-  }, [
-    audience,
-    user?.follower_count,
-    user?.supporter_count,
-    purchasersCount,
-    remixersCount
-  ])
+  const { chatBlastTitle, contentTitle, audienceCount } =
+    useChatBlastAudienceContent({
+      chat
+    })
```

Comment:
love this change!
can reuse the hook for web too?

Reply:
yep! slots in nicely